src/test/java/org/broadinstitute/hellbender/tools/dragstr/InterleavingListSpliteratorUnitTest.java

```diff
@@ -0,0 +1,286 @@
+                }
+                performAdvance(visited, spliteratorStack, times);
+            } else {
+                final Spliterator<Integer> top = spliteratorStack.peek();
+                switch (Operation.of(code)) {
+                    case SPLIT:
+                        Assert.assertFalse(spliteratorStack.isEmpty());
+                        final long topPrevSize = top.estimateSize();
+                        final Spliterator<Integer> other = top.trySplit();
+                        Assert.assertEquals(topPrevSize > 1, other != null);
+                        Assert.assertEquals((other == null ? 0 : other.estimateSize()) + top.estimateSize(), topPrevSize);
+                        if (other != null) {
+                            Assert.assertEquals(((top.estimateSize() - other.estimateSize()) & ~0x01), 0, "unbalanced split"); // checking this subtraction is 0 or 1 by checing every other bit except the first one.
+                            spliteratorStack.pop();
+                            spliteratorStack.push(other);
+                            spliteratorStack.push(top);
+                        }
+                        break;
+                    case ADVANCE:
+                        performAdvance(visited, spliteratorStack, 1);
+                        break;
+                    case READTHRU:
+                        top.forEachRemaining(i -> {
+                            Assert.assertTrue(visited.add(i), "element visited twice: " + i);
+                        });
+                        Assert.assertEquals(top.estimateSize(), 0);
+                        break;
+                    case POPEMPTY:
+                        Assert.assertEquals(spliteratorStack.peek().estimateSize(), 0);
+                        spliteratorStack.pop();
+                        break;
+                }
+            }
+        }
+        Assert.assertTrue(spliteratorStack.isEmpty());
+        Assert.assertEquals(visited.size(), length);
+        Assert.assertTrue(visited.containsAll(subject));
+    }
+
+    private void performAdvance(Set<Integer> visited, Stack<Spliterator<Integer>> spliteratorStack, int times) {
+        long prevSize = spliteratorStack.peek().estimateSize();
+        if (prevSize < times) {
+            Assert.fail("advance beyond end");
+        }
+        for (int i = 0; i < times; i++) {
+            spliteratorStack.peek().tryAdvance(val -> {
+                Assert.assertNotNull(val, "bad argument, null not expected");
+                Assert.assertTrue(visited.add(val), "same element added twice: " + val);
+            });
+            long newSize = spliteratorStack.peek().estimateSize();
+            Assert.assertEquals(newSize, prevSize - 1);
+            prevSize = newSize;
+        }
+    }
+
+    /**
+     * returns a list from 0 to N-1
+     * @param N the lenght of the list.
+     * @return never {@code null}.
+     */
+    private List<Integer> firstNaturalIntegers(final int N) {
```

Comment:
Minor nitpick but this can probably be done with IntStream.range(0,N).Collect(Collectors.toList()) which is a little cleaner